### PR TITLE
doc: Updates the recommended OS to be Ubuntu 22.04

### PR DESCRIPTION
**Pull request by annastuchlik**
Fixes https://github.com/scylladb/scylladb/issues/13138 
Fixes https://github.com/scylladb/scylladb/issues/13153

This PR:

- Fixes outdated information about the recommended OS. Since version 5.2, the recommended OS should be Ubuntu 22.04 because that OS is used for building the ScyllaDB image.
- Adds the OS support information for version 5.2.

This PR (both commits) needs to be backported to branch-5.2.

### CODE REVIEW

**Comment by tzach** [comment]
Lets add 5.2 to the release table as well

**Comment by annastuchlik** [comment]
Sure, I can do it in the same PR.

File: docs/getting-started/os-support.rst [hunk 1]
```diff
@@ -63,17 +61,18 @@ ScyllaDB Open Source
 +----------------------------+------+------+------+------+------+------+------+-------+-------+---------+-------+
 
 
-All releases are available as a Docker container, EC2 AMI, and a GCP image (GCP image from version 4.3).
+All releases are available as a Docker container, EC2 AMI, and a GCP image (GCP image from version 4.3). Since 
+version 5.2, the ScyllaDB AMI/Image OS for ScyllaDB Open Source is based on Ubuntu 22.02.
```
Comment (mykaul) [inline docs/getting-started/os-support.rst:65]:
```suggestion
version 5.2, the ScyllaDB AMI/Image OS for ScyllaDB Open Source is based on Ubuntu 22.04.
```

File: docs/getting-started/os-support.rst [hunk 2]
```diff
@@ -63,17 +61,18 @@ ScyllaDB Open Source
...
 
 ScyllaDB Enterprise
 --------------------
 
 .. note:: 
-   Recommended OS and ScyllaDB AMI/Image OS for ScyllaDB Enterprise:
 
-    - Ubuntu 20.04 for versions 2021.1 and later.
-    - CentOS 7 for versions earlier than 2021.1.
+    The recommended OS for ScyllaDB Enterprise is Ubuntu 22.02.
```
Comment (mykaul) [inline docs/getting-started/os-support.rst:74]:
```suggestion
    The recommended OS for ScyllaDB Enterprise is Ubuntu 22.04.
```

File: docs/getting-started/os-support.rst [hunk 3]
```diff
@@ -95,4 +94,5 @@ ScyllaDB Enterprise
 +----------------------------+------+------+------+------+-------+------+------+------+------+--------+-------+
 
 
-All releases are available as a Docker container, EC2 AMI, and a GCP image (GCP image from version 2021.1).
+All releases are available as a Docker container, EC2 AMI, and a GCP image (GCP image from version 2021.1). Since 
+version 2023.1, the ScyllaDB AMI/Image OS for ScyllaDB Enterprise is based on Ubuntu 22.02.
```
Comment (mykaul) [inline docs/getting-started/os-support.rst:98]:
```suggestion
version 2023.1, the ScyllaDB AMI/Image OS for ScyllaDB Enterprise is based on Ubuntu 22.04.
```

File: docs/getting-started/os-support.rst [hunk 4]
```diff
@@ -25,18 +25,16 @@ ScyllaDB Open Source
 
 .. note:: 
 
-    Recommended OS and ScyllaDB AMI/Image OS for ScyllaDB Open Source:
-
-       - Ubuntu 20.04 for versions 4.6 and later.
-       - CentOS 7 for versions earlier than 4.6.
-
+    The recommended OS for ScyllaDB Open Source is Ubuntu 22.02.
```
Comment (mykaul) [inline docs/getting-started/os-support.rst:28]:
```suggestion
    The recommended OS for ScyllaDB Open Source is Ubuntu 22.04.
```

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please merge to master and backport to branch-5.2.

**Comment by annastuchlik** [comment]
@denesb I can't see the changes on branch-5.2. Could you backport this PR?

I'm sorry for requesting so many backports, but when the new process is in place, these kinds of changes will be made before branching.

**Comment by denesb** [comment]
> @denesb I can't see the changes on branch-5.2. Could you backport this PR?
> 
> I'm sorry for requesting so many backports. When the new process is in place, these kinds of changes will be made before branching.

We can't backport until the patch is not promoted to master.  And I usually forget about a PR after I queue it. So I need a reminder.

**Comment by mykaul** [comment]
Meh - I forgot to comment on Ubuntu 18.0.4. By the time ScyllaDB 5.2 will be released, Ubuntu 18.04 will be very, very near its EOL (April 30, 2023). I see no reason to support it with 5.2. @annastuchlik , @tzach - thoughts?

**Comment by annastuchlik** [comment]
I've moved the discussion to [another issue](https://github.com/scylladb/scylladb/issues/13247).

**Comment by tzach** [comment]
Its better to annoaunce it as depricated in 5.2, and only dropped it in the following release.

**Pull request merged**